kernel-laws/shared/src/test/scala/cats/kernel/laws/LawTests.scala

```diff
@@ -66,8 +66,22 @@ object KernelCheck {
   }
 
   // Copied from cats-laws.
-  implicit def arbitrarySortedSet[A: Arbitrary: Order]: Arbitrary[SortedSet[A]] =
-    Arbitrary(arbitrary[Set[A]].map(s => SortedSet.empty[A](implicitly[Order[A]].toOrdering) ++ s))
+  implicit def arbitrarySortedSet[A: Arbitrary: Order: Cogen]: Arbitrary[SortedSet[A]] = Arbitrary {
+    // We create an arbitrary Ordering[A] which is either the same as Order[A]
+    // or the reverse of it. This is important because the Ordering in use by
+    // a given SortedSet[A] may not be derived from our Order and
+    // scala.math.Ordering does not imply coherence.
+    val ordering: Ordering[A] = Order[A].toOrdering
+    val orderingGen: Gen[Ordering[A]] =
+      Gen.oneOf(Gen.const(ordering),
+                Gen.const(ordering.reverse),
+                Arbitrary.arbitrary[A => Long].map(f => Order.by(f).toOrdering)
+      )
+    for {
+      ordering <- orderingGen
+      sortedSet <- arbitrary[Set[A]].map(s => SortedSet.empty[A](ordering) ++ s)
```

Comment:
it's kind of weird to mix map with for, what about:

```scala
for {
  ordering <- orderingGen
  set <- arbitrary[Set[A]]
} yield (SortedSet.empty(ordering) ++ set)
```
or
```scala
for {
  ordering <- orderingGen
  set <- arbitrary[Set[A]]
  sortedSet = SortedSet.empty(ordering) ++ set
} yield sortedSet
```

Reply:
From my prospective (which can be wrong), `Gen.zip` could be a bit more suitable here:
```
Gen.zip(orderingGen, arbitrary[Set[A]]).map { case (ordering, set) => SortedSet.empty(ordering) ++ set }
```